lib/publiccloud/instance.pm

```diff
@@ -976,4 +976,40 @@ sub wait_for_state {
     die("The instance state is not '$state' but '$current' instead.");
 }
 
+sub check_selinux_pc {
+    # Check SELinux-enforced VMs on publiccloud
+    my ($self) = @_;
+    my $log_avc = "selinux_avc_audit.txt";
+    my $ssh_opts = "-t -o ControlPath=none";
+    die("Unexpected environment.") unless (is_public_cloud && (is_sle_micro || is_sle));
```

Comment:
I think we should not add those restrictions here. If a test calls the library function, one can assume they know the environment.

Also the check for `is_public_cloud` in `lib/publiccloud` appears unnecessary to me.